src/compiler/transformers/generators.ts

```diff
@@ -1124,6 +1124,18 @@ namespace ts {
                 //  .mark resumeLabel
                 //      _b.apply(_a, _c.concat([%sent%, 2]));
 
+                // The `ImportCall` Expression isn't really a call expression, and the import keyword cannot be removed from its arguments
+                if (isImportCall(node)) {
+                    const elements = visitElements(node.arguments);
+                    Debug.assert(elements.kind === SyntaxKind.ArrayLiteralExpression);
+                    // Import call should always have one argument

```

Comment:
This should be an assert. Also, do we need to visit the other elements if we are just going to drop them?

Reply:
It's not an assert because we parse more than one argument for error reporting reasons (and keep them around in the tree) - we just ignore any arguments after the first.

Reply:
And I'm using `visitElements` since it handles local declarations and such potentially required for the argument - recurring into `visitor` directly doesn't do enough :(

Is it worth making a copy of `visitElements` which only visites a single element?

Reply:
`visitElements` is only necessary in `visitCallExpression` because we are passing the output into `Function.prototype.apply`. This is because we capture the target for cases like: `foo(await replaceFooAsync())` where `replaceFooAsync` might replace the value of `foo` at runtime. Since you can't cache `import`, you are probably fine with just calling `visitNode` on the argument and recreating the import call with the result.

Reply:
Actually, all you need to do is change line 1115 (https://github.com/Microsoft/TypeScript/pull/18998/files/5d5fa0d49afb9a4c3b36ed48c8961a10b7eaed6e#diff-7c8a35e1da8c32e5aa5fd22d015c443dR1115) to be `if (!isImportCall(node) && forEach(...)` and let the default behavior of `visitEachChild` below do all the work for you.